src/wallet/wallet.cpp

```diff
@@ -2451,6 +2451,11 @@ static void ApproximateBestSubset(vector<pair<CAmount, pair<const CWalletTx*,uns
                     vfIncluded[i] = true;
                     if (nTotal >= nTargetValue)
                     {
+                        if (fUseInstantSend && nTotal > sporkManager.GetSporkValue(SPORK_5_INSTANTSEND_MAX_VALUE)) {
```

Comment:
It took me some time to understand what this actually does. As I understand it, it skips considering this input as candidate for the "best subset". If this is the intent, wouldn't it be better to just add `if (nTotal + vValue[i].first > max) continue` at the beginning of the inner loop (after line 2441)?